File: 프로그래머스/0/181862. 세 개의 구분자/세 개의 구분자.cs
using System;
using System.Collections.Generic;

public class Solution 
{
    public string[] solution(string myStr) 
    {   
        char[] delimiters = { 'a', 'b', 'c', };
        string[] parts = myStr.Split(delimiters, StringSplitOptions.RemoveEmptyEntries);
        List<string> partsList = new List<string>(parts);
        
        if(partsList.Count == 0) {
            partsList.Add("EMPTY");
        }
        
        parts = partsList.ToArray();
        return parts;
    }
}

File: 프로그래머스/0/181862. 세 개의 구분자/세 개의 구분자.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

vector<string> solution(string myStr) 
{
    vector<string> answer;
    string temp = "";
    for(char &ch : myStr)
    {
        if(ch == 'a' || ch == 'b' || ch == 'c')
        {
            if(!temp.empty())
            {
                answer.push_back(temp);
                temp = "";
            }
        }
        else
        {
            temp += ch;
        }
    }
    if(!temp.empty())
    {
        answer.push_back(temp);
    }
    
    if(answer.empty())
    {
        answer.push_back("EMPTY");
    }
    

    return answer;
}